File: core/server/clientProject/commonClient/src/main/java/com/home/commonClient/scene/unit/CharacterIdentityLogic.java
package com.home.commonClient.scene.unit;

import com.home.commonBase.data.scene.unit.identity.CharacterIdentityData;
import com.home.commonBase.scene.unit.UnitIdentityLogic;

public class CharacterIdentityLogic extends UnitIdentityLogic
{
	private CharacterIdentityData _data;
	
	@Override
	public void init()
	{
		super.init();
		
		_data=(CharacterIdentityData)_unit.getUnitData().identity;
	}
	
	@Override
	public void dispose()
	{
		super.dispose();
		
		_data=null;
	}
	
	/** 获取身份数据 */
	public CharacterIdentityData getData()
	{
		return _data;
	}
}

File: core/client/game/src/commonGame/scene/unit/CharacterIdentityLogic.cs
using System;
using ShineEngine;

/// <summary>
/// 角色身份逻辑
/// </summary>
public class CharacterIdentityLogic:UnitIdentityLogic
{
	private CharacterIdentityData _data;

	public CharacterIdentityLogic()
	{
		
	}

	public override void init()
	{
		base.init();
		_data=(CharacterIdentityData)_unit.getUnitData().identity;
		_unitName=_data.roleShowData.name;
	}

	public override void dispose()
	{
		base.dispose();

		_data=null;
	}
}